### ISSUE TITLE
Alpha Vantage API Review

### ISSUE THREAD

**Issue by wickenden-g** [RomelTorres/alpha_vantage#13]
## Alpha Vantage API Review from a Python Developer

@RomelTorres congratulations on your wonderful python library for Alpha Vantage! 

I am a full-stack python developer who uses the Alpha Vantage API to develop trading strategies for stocks, ETFs, and OTCs. I am wondering whether you are affiliated with Alpha Vantage, and whether this is a good place for me to leave a 360° review on their API service for future reference by the developer community. 

So technically this is not a bug report or feature request - hope it's OK with you :)

### The Experience

I was one of the "yahoo refugees" who stumbled upon Alpha Vantage via Google search. My first interactions with their website could be summarized by the good-old term WYSIWYG (what-you-see-is-what-you-get). Their documentation promised three things: (1) time series data with various resolutions, (2) over 50 technical signals, (3) sector data, and they kept their promise by providing demo URLs for all the said API functions. 

![image](https://user-images.githubusercontent.com/31149405/29483354-48dbc356-8459-11e7-9d0f-e4a3aec08089.png)

They promised free API keys, and they also delivered. They promised "no promotional materials to your inbox," and indeed the only email I got from them so far was the announcement email for their new CSV feature. 

This being said, there are a couple areas they could optimize upon. 
- Be more specific about their business model. Being "100% free" is good, but it can also be a bit scary, especially for a yahoo refugee like me.
- Add multi-symbol support so that one can query multiple stocks with a single API call. 
- Their FAQ recommends "~200 API calls per minute." It would be great if they can set a hard limit on the call volume to prevent client-side abuses in the form of ultra-high-frequency requests. 

### The Data

Of the thousands of US-based equities I have analyzed so far, their historical data and technical indicators seem to match other reputable data sources. Their intraday data is realtime up to the current minute, which is fine for my research purposes but may not satisfy users who want to beat the market with millisecond precision. Perhaps a premium feature for this down the road? 

Their JSON output is easily readable and python-parsable. For the daily time series, however, I understand that their most recent data point is the cumulative information of the current trading day (updated realtime), but why is the first timestamp in YYYY-MM-DD HH:MM:SS format while all the others are in the normal YYYY-MM-DD format typical of the EOD data? 

> "Meta Data": {
>         "1. Information": "Daily Prices (open, high, low, close) and Volumes",
>         "2. Symbol": "MSFT",
>         "3. Last Refreshed": "2017-08-18 16:00:00",
>         "4. Output Size": "Compact",
>         "5. Time Zone": "US/Eastern"
>     },
>     "Time Series (Daily)": {
>         **"2017-08-18 16:00:00"**: {
>             "1. open": "72.2700",
>             "2. high": "72.8400",
>             "3. low": "71.9300",
>             "4. close": "72.4900",
>             "5. volume": "18215276"
>         },
>         **"2017-08-17"**: {
>             "1. open": "73.5800",
>             "2. high": "73.8700",
>             "3. low": "72.4000",
>             "4. close": "72.4000",
>             "5. volume": "21834250"
>         }, 

I would love to see a consistent YYYY-MM-DD format across all the timestamps. The "last refreshed" timestamp can be specified in Meta Data instead:

> "Meta Data": {
>         "1. Information": "Daily Prices (open, high, low, close) and Volumes",
>         "2. Symbol": "MSFT",
>         **"3. Last Refreshed": "2017-08-18 16:00:00"**,
>         "4. Output Size": "Compact",
>         "5. Time Zone": "US/Eastern"
>     },
>     "Time Series (Daily)": {
>         **"2017-08-18"**: {
>             "1. open": "72.2700",
>             "2. high": "72.8400",
>             "3. low": "71.9300",
>             "4. close": "72.4900",
>             "5. volume": "18215276"
>         },
>         **"2017-08-17"**: {
>             "1. open": "73.5800",
>             "2. high": "73.8700",
>             "3. low": "72.4000",
>             "4. close": "72.4000",
>             "5. volume": "21834250"
>         }, 

In addition to the data presentation aspects, below are couple other data-related proposals:
- Expand CSV support to all API functions. CSV is currently enabled only for the time series APIs. 
- Make error messages more informative for debugging purposes. 

### In Summary

It is always a pleasure to have an API service that is well documented, platform/language-agnostic, and easily integratable. The fact that we have several third-party libraries built on top of Alpha Vantage on GitHub is in a sense testament to its developer-friendly nature. While there is still room for them to become a better version of themselves, I hope they thrive and stay true to the description on their home page - "driven by rigorous research, cutting edge technology, and a disciplined focus on democratizing access to data."

**Comment by RomelTorres** [comment]
@wickenden-g Thank you very much for your incredible review. 

Sadly I am not related to Alpha Vantage. One day I also discovered their API while googling and decided to write the python wrapper for it as my first step in playing around with financial algorithms. I am very happy and grateful for the acceptance it has had in the community since this is my first open source project. 

I would contact them directly on their website. They actually write back quickly and are always open to suggestions. Yours seem pretty reasonable. I was thinking on suggesting them on having some sort of company account in Github. 

Thanks again for the 
Romwl

**Comment by frederic-adda** [comment]
Is it only US-stocks?

**Comment by coryleuck** [comment]
@frederic-adda According to my experience with Alpha Vantage, it covers US-based stocks, ETFs, and mutual funds AND stocks traded on major global stock exchanges. Perhaps @RomelTorres could shed some light on this?

**Comment by kevingebert** [comment]
love the Python Wrapper.  Thanks for all the work.  

I am having trouble getting pandas data.  It could be my fault but every time I look at the intraday data for an equity I am missing the bulk of all the data in between.  
For instance, I receive  Oct. 30 data ... Nov 10 data.  but not the rest of the days in between.

**Comment by VJ-Vicky** [comment]
@frederic-adda @coryleuck  No, you It can be used to get data for other stocks as well, for example to get India Stock Exchange listed company TITAN, I have to use _NSE:TITAN_ . This way you can use it the API to get data for Stocks of various Exchanges.

**Comment by VJ-Vicky** [comment]
@frederic-adda @RomelTorres If you want to get data of Stocks from other exchanges from US-Stock exchanges then you need to do something like this, If I wanted to get the stock value for TITAN from Indian Exchange NSE, then I need to use this in symbol **NSE:TITAN** .

Ex:
`ts = TimeSeries(key='API_KEY', output_format='pandas')
data, meta_data = ts.get_daily_adjusted(symbol='NSE:TITAN', outputsize='full')
data['close'].plot()
plt.title('Daily Adjusted for the TITAN stock ')
plt.show()`

**Comment by frederic-adda** [comment]
OK, good to know! Thanks for the info
What I find disturbing is that the ticker symbol are standardized. For example, in the US, the symbol would be like « AAPL », and, for other markets than the US, like  « AI.PA » (where .PA stands for Paris stock exchange).
So AlphaVantage seems to do it differently than the standard way? Weird …
Is there a list of their codes for the Stock exchanges around the world, so I could try to do the matching with the « standard » codes?

Thanks
Fred

**Comment by VJ-Vicky** [comment]
@frederic-adda I doubt that there is an API which will provide you the list of all Stock Exchanges around the world. 

But still are here some links that might help you out, http://www.wikinvest.com/wiki/List_of_Stock_Exchanges
http://www.eoddata.com/

**Comment by frederic-adda** [comment]
Yahoo Finance used to do it. 
I’m looking for a replacement for their API.

**Comment by kevingebert** [comment]
HI, great python framework.  Thank you so much for putting it together.  I would love to see the adjusted data as an option

**Comment by RomelTorres** [comment]
Hi @kevingebert; what do you mean by adjusted data? Any call from the API I am missing?

**Comment by kevingebert** [comment]
Maybe I am missing it.  I'm talking about the split adjusted data.

**Comment by samperd** [comment]
@RomelTorres , 

Great work on the API wrapper. I use the yahoo python wrapper ystocks a number of years ago. now that I have returned to a financial project I learned about the changes at yahoo. And then i found Alpha Vantage and then your API.

I will poke through the basics and see what is there right now.

The first thing I have not been able to find is the dividend event data. See issue #27

Great work so far.

**Comment by vignesh989** [comment]
Hi Guys, 
@RomelTorres,@wickenden-g
This is a really amazing API, now that Google API is no longer supporting APIs. This seems to be the next best thing. 
I have two major concerns, I'm creating a Machine Learning algorithm to predict trends. So for this, I require Intraday data ( 5Mins ) for a stock for the past let's say 200 days, as a training set for my algorithm. 

1. Is there a way to specify past x days, or specify the date on which I want the intraday data for ?

2. @VJ-Vicky What seems to be wrong with my query? 
https://www.alphavantage.co/query?function=TIME_SERIES_INTRADAY&symbol=NSE:TITAN&outputsize=compact&interval=5min&datatype=json&apikey=Key
The Solution you suggested for the specifying the EXchange is not working it is returning me an error.

**Comment by RomelTorres** [comment]
Hi @vignesh989 if it is not specified in the API by alphavantage, it is not doable direct in the python wrapper, nevertheless once you have the data in a pandas data frame. It should be easy to slice it. I would say first do the intraday call with full size and then slice it from there.

**Comment by vignesh989** [comment]
@RomelTorres Thank you, I get a max for past 3 days even if I get the full size, I guess I should try a different data source.

**Comment by ellissitzky** [comment]
Hi, I am currently developing a basic Java (Android)  app for portfolios management. Alpha Vantage data are very good for my purpose. I am wondering if i can load this applet on GooglePlay (free with little advertising) each user should request his own ApiKey. Could it be compatible with the terms of usage?Thank you.

**Comment by RomelTorres** [comment]
Hi @ellissitzky. That I cannot answer you, since here in github I just habe the python wrapper. Please send an email to the people from alphavantage on their website.

Thanks,

Romel

**Comment by ellissitzky** [comment]
@RomelTorres, thank you for your answer, i first asked to this blog in order to know if this info were already clear.. I read everything on AlphaVantage site, but the only part that may be about the topic is : https://www.alphavantage.co/terms_of_service/ - Intellectual Property. It's no clearly expressed a prohibition of use in apps, etc, but not even an encouragement to it. So, I will ask  them by email . 
Thanks,
Giuseppe

**Comment by team-alphavantage** [comment]
@ellissitzky Thank you for developing the Java application on top of Alpha Vantage. We have addressed your request via support@alphavantage.co

**Comment by ellissitzky** [comment]
@team-alphavantage,  I just read (and answered) it! 
Thanks, Great!

**Comment by Ham-Merhead** [comment]
A wonderful piece of work, thank you.
I perhaps found a problem with pricing Bitcoin in USD. The problem does not occur on the AV HTML interface, and USD is included in https://www.alphavantage.co/physical_currency_list. 

get_digital_currency_daily(symbol='BTC',market='CNY')      # WORKS OK
get_digital_currency_daily(symbol='BTC',market='GBP')      # WORKS OK
get_digital_currency_daily(symbol='BTC',market='USD')      # FAILS, message is:
                   Alpha Vantage Interface Exception cannot convert the series to <class 'float'>
BUT this works:
https://www.alphavantage.co/query?function=DIGITAL_CURRENCY_DAILY&symbol=BTC&market=USD&apikey=MY_KEY

Btw, all markets also return a price in USD, but this varies significantly between markets. The volume on the USD market appears much larger and perhaps is a more reliable price.

**Comment by Ham-Merhead** [comment]
Wonderful piece of work, thank you.

One small issue using "get_digital_currency_daily()". The result is returned in two currencies, market and USD, OK. But if I set the market to USD I get duplicate labels.

So indexing a row ( row['. close (USD)'] ) results in a two element series for USD, and a scalar for other markets. I can special case USD, but it would be nice if you did.

**Comment by RomelTorres** [comment]
Hi @Ham-Merhead, I will have to check it out, I have been super busy lately and have not had a chance to sit down and solve some of the issues that have arisen in the past few weeks. Could you please post this in a separate issue with the piece of code and the resulting pandas data frame? 

I am planning to do a rework of the unittests, since they are not working well anymore, although most of the calls are correct (because I was a noob and did not know better when I wrote that part of code)

I will try to get some free time and sit and do it properly. 

Saludos Romel

**Comment by Ham-Merhead** [comment]
Done. Thanks for the the response. I have a workaround for now. Interesting question as to if any change would happen here or by AV.......

**Comment by igorcc** [comment]
What's about Swiss Exchange (SIX) ? I tried to request SIX:NESN but with no success.. Is there a way to request data from SIX, LSE, Xetra? 
Thanks!

**Comment by ChillarAnand** [comment]
@vignesh989 Did you manage to get data from NSE? I am not able to pull the data. Here is the sample code. 

```py
import os

from alpha_vantage.timeseries import TimeSeries


key = os.environ.get('ALPHAVANTAGE_KEY')
print(key)
ts = TimeSeries(key=key)

data, meta_data = ts.get_intraday('NSE:TITAN')
# data, meta_data = ts.get_intraday('NSE:RELIANCE')
# data, meta_data = ts.get_intraday('GOOGL')
print(meta_data)
```

**Comment by igorcc** [comment]
I cannot retrieve data for NSE. Just TITAN works well. 
NSE:TITAN produces following error: "ValueError: Invalid API call. Please retry or visit the documentation (https://www.alphavantage.co/documentation/) for TIME_SERIES_INTRADAY."

**Comment by RomelTorres** [comment]
Hi people, you can try directly in a web browser and see if it works. If it does create a new issue and I will work on it. If not we can ask in the alphavantage forum.

**Comment by igorcc** [comment]
Browser produces the same error. I think the problem is in alphavantage data.

**Comment by vignesh989** [comment]
@ChillarAnand @igorcc No Guys they do not support it as of now, So this is workaround. So you can determine if the data is coming from NYSE or NSE the following way. 
You can convert date column to your local time. 
`DateTime estTime = Convert.ToDateTime(item.Key.ToString());`
` TimeZoneInfo estZone = TimeZoneInfo.FindSystemTimeZoneById("Eastern Standard Time");`
`TimeZoneInfo istZone = TimeZoneInfo.Local;`
`DateTime yourLocalTime = TimeZoneInfo.ConvertTime(estTime, estZone, istZone);`

If the market Start Time/End Time syncs with your market time it means its from your market, else its from NYSE.

![image](https://user-images.githubusercontent.com/8469272/34843231-e36e36ae-f733-11e7-970c-1e4c421b1c76.png)
If you look at the image above you will see that the closing time for HDFC ( NSE ) and INFY ( NYSE ) is different. If you convert it to IST it translates to 3:30 PM ( NSE closing time )

So the best bet for people who are working with other markets is ignore stocks that have the same symbol as the stocks in NYSE. 

![image](https://user-images.githubusercontent.com/8469272/34843446-8844b2d4-f734-11e7-9373-ba272ff6bad2.png)

@ChillarAnand this is still a very good set of stocks, in NIFTY50 only 3 gets eliminated this way, rest of the stock data works as expected. 

@team-alphavantage It would be wonderful if you can add a parameter to your API to specify the market.

@RomelTorres I'm so sorry to hijack your thread like this.

**Comment by sukhdev22** [comment]
Does it provide futuers and options data? If yes, How do I get symbols for it?

**Comment by me-shivaprasad** [comment]
@vignesh989 @RomelTorres Is there is the way to extract the NIFTY  and SENSEX data ex Index price

**Comment by raghav1320** [comment]
Can we change the Time Zone in alphavantage ?

**Comment by rollnlearn** [comment]
How do we get stock and indices ticker symbol for Indian Stock Exchanges. Nothing is mentioned related to that in the documentation. What is the ticker symbol I need to to get data for infosys stock in National Stock Exchange (NSE) and Yes Bank stock in Bombay Stock Exchange (BSE).

**Comment by vignesh989** [comment]
@rollnlearn I think there was a recent update, not 100% sure. But now I'm able to specify the market details: for example, if I need Infosys from (NSE) I symbol=INFY.NS. if I want from (BSE).  symbol=INFY.BO. 

But then again it does not work on all the stocks, I found it pulls an errors some stocks like L&T.
Even if a market is specified. 
You can give it a try. Let us know if you are able to identify the issue.

@raghav1320  No, you will not be able to change the time zone. The data is always given in EST, you can convert it to your respective time zone via code. 

@me-shivaprasad No, I'm not aware of any such option.

**Comment by meaninglessLives** [comment]
@rollnlearn You can find the ticker index for NSE here [https://www.nseindia.com/corporates/corporateHome.html?id=equity](url)

**Comment by gautamlaungani** [comment]
for NSE use the Yahoo Codes or for any ticker in the world, Yahoo codes work.

eg to download csv files

https://www.alphavantage.co/query?function=TIME_SERIES_DAILY_ADJUSTED&symbol=EICHERMOT.NS&apikey="Your_API_Key"&datatype=csv&outputsize=full
--
https://www.alphavantage.co/query?function=TIME_SERIES_DAILY_ADJUSTED&symbol=HEROMOTOCO.NS&apikey="Your_API_Key"&datatype=csv&outputsize=full
https://www.alphavantage.co/query?function=TIME_SERIES_DAILY_ADJUSTED&symbol=M&M.NS&apikey="Your_API_Key"&datatype=csv&outputsize=full
https://www.alphavantage.co/query?function=TIME_SERIES_DAILY_ADJUSTED&symbol=MARUTI.NS&apikey="Your_API_Key"&datatype=csv&outputsize=full
https://www.alphavantage.co/query?function=TIME_SERIES_DAILY_ADJUSTED&symbol=TATAMOTORS.NS&apikey="Your_API_Key"&datatype=csv&outputsize=full


the adjustments can be done via the formula 

Adj Open = Open / (Close/AdjustedClose)
Adj High = High /  (Close/AdjustedClose)
Adj Low = Low /  (Close/AdjustedClose)

Volume is adjusted for Splits

i have back tested the results with data from Exchange Data International (premium service) and post the adjustment, the data is almost 99% accurate for Open/High/Low.

if anyone cares for a VBA code which can be used to cure data for x number of csv files in a particular folder, hit me up.

**Comment by Prchakr** [comment]
I am not sure what is the issue with NSE market.
As far i read i see issue with getting intraday quote for NSE market i am able to fetch it without any issue
I am trying something like below & works fine for me
https://www.alphavantage.co/query?function=TIME_SERIES_INTRADAY&symbol=SBIN&interval=1min&apikey=APIKey
tried other symbold like GAIL, YESBANK etc., & it looks good.

tried all the combinations to fetch batch quote & none of it working,
https://www.alphavantage.co/query?function=BATCH_STOCK_QUOTES&symbols=SBIN&apikey=APIKEY
https://www.alphavantage.co/query?function=BATCH_STOCK_QUOTES&symbols=NSE:SBIN&apikey=APIKEY
https://www.alphavantage.co/query?function=BATCH_STOCK_QUOTES&symbols=SBIN,GAIL&apikey=APIKEY

My Output for batch quote is as follows, 

{
    "Meta Data": {
        "1. Information": "Batch Stock Market Quotes",
        "2. Notes": "IEX Real-Time Price provided for free by IEX (https://iextrading.com/developer/).",
        "3. Time Zone": "US/Eastern"
    },
    "Stock Quotes": []
}


And, this batchquote works well for USstocks, i tried with AAPL & output is as expected.
I belive batch quote is not supported for all the markets.
Anyway, if anyone got luck, please post it.

**Comment by feresm** [comment]
Is there any way I can get funds from their ISIN using alpha vantage?

**Comment by HyperionSounds** [comment]
Hi Romel, 

This is just what I've been looking for. And as someone that's relatively new to python, the documentation is extremely appreciated.

My best regards.

**Comment by anniew86** [comment]
@igorcc For Swiss stock use suffix VX, like NESN.VX

**Comment by anniew86** [comment]
oops, there are huge data gaps with missing data, do you know any reason for it?

https://www.alphavantage.co/query?function=TIME_SERIES_DAILY&symbol=NESN.VX&apikey=XXX

```
"Meta Data": {
        "1. Information": "Daily Prices (open, high, low, close) and Volumes",
        "2. Symbol": "NESN.VX",
        "3. Last Refreshed": "2018-08-08",
        "4. Output Size": "Compact",
        "5. Time Zone": "US/Eastern"
    },
    "Time Series (Daily)": {
        "2018-08-08": {
            "1. open": "81.8600",
            "2. high": "82.0600",
            "3. low": "81.7000",
            "4. close": "81.8600",
            "5. volume": "2758215"
        },
        "2018-06-12": {
            "1. open": "74.3600",
            "2. high": "74.8000",
            "3. low": "74.2800",
            "4. close": "74.5400",
            "5. volume": "6568048"
        },
        "2018-06-11": {
            "1. open": "73.7600",
            "2. high": "74.5000",
            "3. low": "73.7200",
            "4. close": "74.2000",
            "5. volume": "5416673"
        },
...
```

**Comment by tarioch** [comment]
You can also use .SW suffix which for Swiss stocks and from a quick search looks like SW is the better one: https://assets.bwbx.io/documents/users/iqjWHBFdfxIU/rH89yROZwbeA/v0

**Comment by anniew86** [comment]
not for Alphavantage, NOVN.SW does'n work:
https://www.alphavantage.co/query?function=TIME_SERIES_DAILY&symbol=NESN.SW&apikey=XXX

**Comment by tarioch** [comment]
Looks like for Stocks it's VX and for others such as ETFs it's SW, e.g. 
https://www.alphavantage.co/query?function=TIME_SERIES_DAILY&symbol=CSBGC7.SW&apikey=XXX

**Comment by dokmaister** [comment]
@anniew86: .SW and .VX are for Swiss stocks whereas .VX is for blue chips like Roche, Nestlé etc and .SW is for mid- and small caps and ETFs. It's the same on yahoo (e.g. NOVN.VX vs. KARN.SW). VX stands for former VIRT-X I believe. So use VX for SMI Index companies and SW for the rest, think that should work.

**Comment by davepaiva** [comment]
Hey guys, if you'll are having trouble with localizing the timezone of the data, check out my code stock_data_alphavantage.py 

https://github.com/davepaiva/StockAnalysis

**Comment by sanjes1** [comment]
Hi, Just started to use alphavantage.
Trying 
https://www.alphavantage.co/query?function=TIME_SERIES_INTRADAY&symbol=MSFT&interval=1min&outputsize=full&apikey="MyKey"

It only returns few days of data and not all. 
Is there a limit or am I missing something.
Any help is appreciated. 
Thanks

**Comment by pshirlyn** [comment]
There is a soft limit on intraday calls to make the amount of data reasonably efficient, but you can always reach out to them on their website if you need more data! If you do not need 1-minute interval data points, you should use the daily time series (TIME_SERIES_DAILY) instead.

**Comment by sanjes1** [comment]
Thanks for the response.  I did tried sending an email to them, no response yet.
I will need the 1 minute data. 
Not sure if i will get the data by subscribing to the premium key.

**Comment by mark100net** [comment]
@sanjes1 They have never responded to me. They don't believe in support apparently.

**Comment by SterlingButters** [comment]
> 1. Is there a way to specify past x days, or specify the date on which I want the intraday data for?

I too would much like to see this since even though panda df slicing isnt a problem, trying to download a mass of data for one data point is quite expensive. This the most anticipated prospective feature of the Alpha Vantage API for me

**Comment by sanjes1** [comment]
Ok, so they will talk, but you need to pay them. I subscribed to their monthly plan. They responded pretty quick.
Also for my question they provide last 1 week~ of intraday 1 min data. I dont see any interface to request for a particular date

**Comment by SterlingButters** [comment]
@sanjes1 The "~" 1 week of intraday depends on how many days the market is open out of the that last 7 I believe. I would use your premium status to request features such as that if you haven't already.

**Comment by sanjes1** [comment]
@SterlingButters 
I tried, They said, they are working on it to get more data and dont have any dates :)

**Comment by ramkoduru** [comment]
Hi, Is there any way where we can get information of the corresponding sector for the stocks we get the data or any way we can pull the stock performance(API available) along with corresponding industries. i am working on stock predictions so checking i may get any help on this

**Comment by StevenWilliams** [comment]
What's the split coefficient supposed to be? I checked for some stocks that split and it was always 1.00

**Comment by hsunesen** [comment]
For me it has worked, using this formel for symbol, for european stocks:

Fingerprint Cards (STO: FING-B) == FING-B.STO 
Novo Nordisk (CPH: NOVO-B) == NOVO-B.CPH
BMW (DE: BMW) == BMW.DE

I think you got the point ;)

**Comment by PushT** [comment]
If AlphaVantage are rearranging the letters in the acronyms of the stocks on different exchanges(for their API calls), they have to have a list clarifying the logic behind that. There IS no logic otherwise.

**Comment by BipinSha** [comment]
> NSE:TITAN

@RomelTorres  @thecodeflash, are you guys getting the correct timestamp for NSE:TITAN or any other NSE stock, i think it should show till 3:30 PM or something, it is showing 4:59.

**Comment by PatrickAlphaVantage** [comment]
> > NSE:TITAN
> 
> @RomelTorres @thecodeflash, are you guys getting the correct timestamp for NSE:TITAN or any other NSE stock, i think it should show till 3:30 PM or something, it is showing 4:59.

If you look at the time zone, it's in `6. Time Zone: "US/Eastern"` you'll have to convert it to IST if that's what you're looking for. 
Check out this [thread in StackOverflow](https://stackoverflow.com/questions/58654008/how-to-change-time-zone-in-alpha-vantage-api/58662045?noredirect=1#comment103707403_58662045) that answers the question.

**Comment by jcisio** [comment]
Hello,

I know this is a long thread with a lot of questions. I should not add another comment, but I will do now, because there is a repeated question without a real question.

To get the symbol of the instrument, you can search using ISIN. For example `https://www.alphavantage.co/query?function=SYMBOL_SEARCH&keywords=FR0000120172&apikey=demo`

```json
{
    "bestMatches": [
        {
            "1. symbol": "CA.PAR",
            "2. name": "Carrefour SA",
            "3. type": "Equity",
            "4. region": "Paris",
            "5. marketOpen": "09:00",
            "6. marketClose": "17:30",
            "7. timezone": "UTC+01",
            "8. currency": "EUR",
            "9. matchScore": "0.1667"
        }
    ]
}
```

**Comment by spateltc** [comment]
how to fetch data for nifty 50?can anyone help me with the url which fetch nifty 50 data

**Comment by PatrickAlphaVantage** [comment]
Hi @spateltc,

Currently, Alpha Vantage has support for stocks, crypto, TIs, sector, and FX. Index data you can get from pulling the individual stocks in the nifty 50.

**Comment by jcisio** [comment]
> how to fetch data for nifty 50?can anyone help me with the url which fetch nifty 50 data

The symbol is `NFTY`. Use the search function as I pointed out in the previous comment to find the symbol. The first result is the correct one.

```
https://www.alphavantage.co/query?function=SYMBOL_SEARCH&keywords=NIFTY&apikey=${"apikey"}

{
    "bestMatches": [
        {
            "1. symbol": "NFTY",
            "2. name": "First Trust India NIFTY 50 Equal Weight ETF",
            "3. type": "ETF",
            "4. region": "United States",
            "5. marketOpen": "09:30",
            "6. marketClose": "16:00",
            "7. timezone": "UTC-05",
            "8. currency": "USD",
            "9. matchScore": "0.8889"
        },
        {
            "1. symbol": "ICICINIFTY.BSE",
            "2. name": "ICICI Prudential Nifty ETF",
            "3. type": "Equity",
            "4. region": "India/Bombay",
            "5. marketOpen": "09:15",
            "6. marketClose": "15:30",
            "7. timezone": "UTC+5.5",
            "8. currency": "INR",
            "9. matchScore": "0.5556"
        }
    ]
}
```
Then fetch the data:
```
https://www.alphavantage.co/query?function=GLOBAL_QUOTE&symbol=NFTY&apikey=${"apikey"}

{
    "Global Quote": {
        "01. symbol": "NFTY",
        "02. open": "34.8600",
        "03. high": "34.8600",
        "04. low": "34.8600",
        "05. price": "34.8600",
        "06. volume": "6",
        "07. latest trading day": "2020-01-06",
        "08. previous close": "34.9252",
        "09. change": "-0.0652",
        "10. change percent": "-0.1867%"
    }
}
```

**Comment by jcisio** [comment]
Please ignore my previous comment. It was about an ETF. As per @PatrickAlphaVantage 's answer, to calculate an index, calculate the components is a solution.

**Comment by PatrickAlphaVantage** [comment]
Woah that’s gotta be new. Looks like we must have added this recently, that’s great! Thanks @jcisio !

**Comment by PatrickAlphaVantage** [comment]
Thanks @jcisio for the info on the ETF

**Comment by jcisio** [comment]
@PatrickAlphaVantage I was looking for an alternative to my costly provider and alphavantage was one of the candiates but it has some blocking problems. Now I've just found out another big problem: no index data :( I've just searching a few examples and it really does not have index data. We can also use maths to calculate, but then it adds extra complexity.

**Comment by dominhtri1995** [comment]
> Ok, so they will talk, but you need to pay them. I subscribed to their monthly plan. They responded pretty quick.
> Also for my question they provide last 1 week~ of intraday 1 min data. I dont see any interface to request for a particular date

You can use Finnhub instead. [](https://finnhub.io/docs/api#stock-candles) . They will let you specify from and to timestamp with 1 minute historical data going back for 20 years. 60 free API calls/minute as well.



> I was looking for an alternative to my costly provider and alphavantage was one of the candiates but it has some blocking problems. Now I've just found out another big problem: no index data :( I've just searching a few examples and it really does not have index data. We can also use maths to calculate, but then it adds extra complexity.

Again, using Finnhub, you can pull quote of global index like this:
[](https://finnhub.io/api/v1/quote?symbol=^NIFTY)
[](https://finnhub.io/api/v1/quote?symbol=^GSPC) (S&P 500)
[](https://finnhub.io/api/v1/quote?symbol=^DJI) (Dow)

**Comment by PatrickAlphaVantage** [comment]
@dominhtri1995 just a reminder that this is a thread about helping developers with questions and reviews of Alpha Vantage. 

AV has clean, high quality, correctly adjusted price data.

**Comment by dattran2346** [comment]
> @sanjes1 You can use Finnhub instead. https://finnhub.io/docs/api#stock-candles . They will let you specify from and to timestamp with 1 minute historical data going back 20 years. 60 free API calls/minute as well.

@dominhtri1995, Thank you !!!! Their intraday data indeed goes back to at least 2000 (I haven't tested further than that). It's nice that we can stream tick data with Finnhub API via their websocket as well. Market-wide 1 minute bars are usually extremely expensive but they are offering it for free. 

I do think that this is an open source project, not a commercial platform of alpha vantage. Hence, IMHO, we should encourage all contributions and suggestions which benefit the community.

**Comment by PatrickAlphaVantage** [comment]
@dattran2346 Well said. 
I was referring more to the topic of the thread "Alpha Vantage API Review", but I guess it did sort of turn into a general discussion room haha.

**Comment by wickenden-g** [comment]
Hi folks, I am the OP of this Github issue. Appreciate the responses to my post (my inbox is exploding!), but frankly I don't feel comfortable seeing my review article morphing into tangential discussions. This issue is for reviewing the **Alpha Vantage APIs**, not for technical support or marketing campaigns. 

Can the moderators do something about the off-topic posts?

**Comment by steveotistrader** [comment]
Noob here. Been testing the two. Alpha Vantage has been really solid so far. These other guys I can barely get any data back. And it looks wrong. 
<img width="1423" alt="Screen Shot 2020-01-23 at 1 26 26 PM" src="https://user-images.githubusercontent.com/60218273/73012536-8d426d00-3de4-11ea-87ac-cc85be73d7e9.png">

**Comment by PatrickAlphaVantage** [comment]
Agreed, thanks @wickenden-g . As we are big fans of open source, we want to foster a community that is open to discussion, however I think you make an even better point that the thread has devolved. 

We will close the thread but not lock it, so discussion may continue, it just won't show up as an issue anymore. For more discussion, please feel free to check out the [Alpha Vantage slack](https://alphavantage.herokuapp.com/)
Again a huge thanks to @wickenden-g for making this awesome review, and @RomelTorres for this awesome open source project!

**Comment by tmd2142** [comment]
Wow. just wow. A whole long discussion off topic was fine. The moment someone mention an alternative, all of a sudden the thread is closed and the author is "uncomfortable" lol. This mindset is extremely dangerous for the open source community.

**Comment by jcisio** [comment]
@tmd2142 please respect the OP and the project developer (I guess that Patrick has joined this project). Off-topic discussion should stop, late better than never.

Besides, I don't think a group of people just come and advertise another solution is healthy for the community. Unless the topic is for that, which is not the case here. I won't bother discuss about Finnhub (I did see some problems with it) because it will make this even more off-topic.

**Comment by tmd2142** [comment]
> @tmd2142 please respect the OP and the project developer (I guess that Patrick has joined this project). Off-topic discussion should stop, late better than never.
> 
> Besides, I don't think a group of people just come and advertise another solution is healthy for the community. Unless the topic is for that, which is not the case here. I won't bother discuss about Finnhub (I did see some problems with it) because it will make this even more off-topic.

Don't get me wrong. I'm a fan of alpha vantage and been using this library for a long time now. Just saying that we shouldn't turn this into a place where we shut down all alternatives which might or might not benefit the community.

I think this discussion is really off topic now so I will stop.
